### PR TITLE
[connectors] fix: Improved error handling on files sync

### PR DESCRIPTION
**Pull request by tdraier**
## Description

handle* methods return Result object with a CoreAPIDataSourceDocumentSection or an error object, that can be handled in the activity.

## Deploy Plan

Deploy connector

### CODE REVIEW

File: connectors/src/connectors/google_drive/temporal/file.ts [hunk 1]
```diff
@@ -145,8 +145,13 @@ async function handleFileExport(
       connectorId,
     });
   } else {
-    return handleTextExtraction(res.data, localLogger, file.mimeType);
+    result = await handleTextExtraction(res.data, localLogger, file.mimeType);
+  }
+  if (result.isErr()) {
```
[the remark this answers is not in the document]
Comment (tdraier) [reply connectors/src/connectors/google_drive/temporal/file.ts:150]: It's already logged in all handle* methods higher , I kept the same behaviour (previously no log, just return the result)

File: connectors/src/connectors/microsoft/temporal/file.ts [hunk 2]
```diff
@@ -282,17 +214,73 @@ export async function syncOneFile({
...
+        : null;
+    } else {
+      localLogger.info(
+        {
+          documentLen: documentLength,
+        },
+        `Document is empty or too big to be upserted (marking as synced without upserting)`
+      );
+
+      result = new Err(new Error("document_too_big_or_empty"));
```
Comment (flvndvd) [inline connectors/src/connectors/microsoft/temporal/file.ts:273]: We shouldn't mutate result like this. Can we simply return here?

File: connectors/src/connectors/microsoft/temporal/file.ts [hunk 3]
```diff
@@ -282,17 +214,73 @@ export async function syncOneFile({
...
+    }
+  }
+
   if (fileResource) {
     await fileResource.update(resourceBlob);
```
Comment (flvndvd) [inline connectors/src/connectors/microsoft/temporal/file.ts:278]: So we don't differentiate successfully imported than the one that failed to be sync. This seems to be an issue as those docs will show up in the the assistant builder whereas they don't have any data.

**Review by flvndvd — APPROVED** [review]
LGTM left a few small nits, but feel free to ship once addressed.

**Pull request merged**